master/internal/rm/multirm/multirm.go

```diff
@@ -0,0 +1,370 @@
+package multirm
+
+import (
+	"crypto/tls"
+	"fmt"
+
+	"github.com/labstack/echo/v4"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/maps"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/rm"
+	"github.com/determined-ai/determined/master/internal/rm/agentrm"
+	"github.com/determined-ai/determined/master/internal/rm/kubernetesrm"
+	"github.com/determined-ai/determined/master/internal/rm/rmerrors"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/aproto"
+	"github.com/determined-ai/determined/master/pkg/command"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/jobv1"
+)
+
+// ErrRMNotDefined returns a detailed error if a resource manager isn't found in the MultiRMRouter map.
+func ErrRMNotDefined(rm string) error {
+	return fmt.Errorf("resource manager %s not defined", rm)
+}
+
+// MultiRMRouter tracks all resource managers in the system.
+type MultiRMRouter struct {
+	defaultRMName string
+	rms           map[string]rm.ResourceManager
+	syslog        *logrus.Entry
+}
+
+// New creates a new resource manager with the master set-up.
+// Since multirm imports the rm module, we have to put the New() func
```

Comment:
i have some more vague feedback about how to move this around so you dont mess up imports but also leave rm.New() as what is called.